fiatconnect-api.md

```diff
@@ -391,105 +407,204 @@ The FiatConnect API specifies *two* types of authentication; the first authentic
+On failure, the server MUST return an HTTP `401` error, along with an error code that details exactly why the request failed.
 
-##### 3.2.1.2.3. `"exp"` Claim
+###### 3.2.1.5.3.2.1 `InvalidSignature`
 
-The `exp`, or *expiration time* claim represents the time until which the provided JWT should be accepted by the server. The `exp` claim should be provided
-as a numeric timestamp, defined as the number of seconds since Epoch. The `exp` claim MUST be present for all endpoints *except* the following:
+If the signature is invalid or doesn't match the included `address` field, the server MUST return an `InvalidSignature` error. Note that the address used
+to sign the message may not actually match the address being authenticated in the case where the client is authenticating as a contract-owned account. In
+such cases, `InvalidSignature` MUST NOT be returned when the signature is valid, and the address used to sign the message is considered valid by the contract
+referenced in the `address` field.
+
+###### 3.2.1.5.3.2.2 `InvalidParameters`
+
+If the request body is missing, malformed, or fails to pass any of the required checks, the server MUST respond with an `InvalidParameters` error.
+
+#### 3.2.1.6. Using Sessions
+
+Once a client has established a session with a server, they can use that session (by sending the session cookie to the server) to access priveleged endpoints
+throughout the API. Certain endpoints impose different requirements on the in-use session. There are *two levels* of endpoints, with respect to their authentication
+requirements. These levels are explained below.
+
+##### 3.2.1.6.1. Non-Privileged Endpoints
+
+The first group of endpoints are those that *do not* require the user to have a logged-in session. These are:
 
 * `GET /quote/in`
 * `GET /quote/out`
+* `POST /auth/login`
+
+Since it may be a painful user experience to have a user log in to a provider just to get a quote, authentication is not required for these endpoints. Of course,
+clients may still access these endpoints when logged in, but a logged-in session (or any session at all) is not required to access these endpoints. `POST /auth/login`
+does not require a user to already be logged in, as described above.
+
+##### 3.2.1.6.2. Privileged Endpoints
+
+The second, and largest set of endpoints, are the ones that require a user to be logged in. These endpoints are the ones that deal with reading and writing user-specific data.
+
+* `POST /kyc/:kycSchema`
 * `GET /kyc/:kycSchema/status`
+* `DELETE /kyc/:kycSchema`
+* `POST /accounts/:fiatAccountSchema`
+* `GET /accounts`
+* `DELETE /account/:fiatAccountId`
 * `GET /transfer/:transferId/status`
+* `POST /transfer/in`
+* `POST /transfer/out`
+
+For these endpoints, a user needs to be logged in with an authenticated session. This session may have the maximum expiration time allowed by the FiatConnect standard (four hours).
```

Comment:
```suggestion
For these endpoints, a user needs to be logged in with an authenticated session.
This session MUST have an expiration time of no greater than four hours.
```

Just to make it a bit more clear that this is a requirement (I also foresee service not honoring it, but what can you do 🤷‍♂️)